.github/workflows/publish-mac-arm.yml

```diff
@@ -0,0 +1,59 @@
+name: Publish MacOS-ARM release
+
+on:
+  workflow_dispatch:
+  
+permissions:
+  id-token: write 
+  contents: write
+
+jobs:
+  get_release_version:
+    name: Get Release Version
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'ballerina-platform'
+    steps:
+      - id: checkout
+        uses: actions/checkout@v2
+      - id: execute
+        run: |
+          export VERSION="$((grep -w 'version' | cut -d= -f2) < gradle.properties | cut -d- -f1)"
+          echo "::set-output name=version::${VERSION}"
+    outputs:
+      version: ${{steps.execute.outputs.version}}
+  update_release:
+    needs: get_release_version
+    name: Update Release
+    runs-on: ubuntu-latest
+    if: github.repository_owner == 'ballerina-platform'
+    outputs:
+      upload_url: ${{ steps.update_release.outputs.upload_url }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
+      - name: cosign-installer
+        uses: sigstore/cosign-installer@v3.0.3
+      - name: Retrieve MacOS-ARM Artifact
+        run:
+          |
+          wget https://github.com/${{ github.actor }}/ballerina-custom-jre/releases/download/${{ needs.get_release_version.outputs.version }}/ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.zip
+      - name: Sign the MacOS-ARM Artifact
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+        run: |
+          cosign sign-blob ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.zip --output-certificate ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.pem --output-signature ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.sig --yes
+      - name: Verify the MacOS-ARM Artifact
+        env:
+          COSIGN_EXPERIMENTAL: "true"
+        run: |
+          cosign verify-blob ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.zip --certificate ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.pem --signature ballerina-jre-macos-arm-64-${{ needs.get_release_version.outputs.version }}.sig --certificate-identity=https://github.com/${{ github.actor }}/ballerina-custom-jre/.github/workflows/publish-mac-arm.yml@refs/heads/main --certificate-oidc-issuer=https://token.actions.githubusercontent.com 
+      - uses: IsaacShelton/update-existing-release@v1.3.2
```

Comment:
@Miranlfk Shall we use an official action like ` actions/upload-release-asset`

Reply:
I'll make the change

Reply:
The official action cannot be used in the case of updating an existing release, as per my testing. I've opted to use `github-cli` to do the functionality